### PR TITLE
[NEW] Matrix federation events coverage expansion + bug fixes

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
## Proposed changes (including videos or screenshots)

- Disables not supported features in federated channels (#27204)
- Add support for the whole Matrix Markdown spec (#27725)
- Allow federated rooms with the same name per different servers when they have a contact between them (#27559)
-  Matrix Federation send message when multiple users are involved was not working (#27327)
- Allow the room owner to set roles on federated rooms, following a specific and different approach as normal rooms (available on docs) (https://github.com/RocketChat/Rocket.Chat/pull/27633)
- E2E Tests (PS: they don't run on CI) (#27578 #27282 #27115)
- Disables admin actions disallowed for federated users/rooms (https://github.com/RocketChat/Rocket.Chat/pull/27251)

FED-81

### CODE REVIEW

File: apps/meteor/app/lib/server/functions/deleteUser.ts [hunk 1]
```diff
@@ -15,19 +16,15 @@ import { LivechatUnitMonitors } from '../../../../ee/app/models/server';
...
 	}
 
-	if (user.federation) {
-		const existingSubscriptions = Subscriptions.find({ 'u._id': user._id }).count();
-
-		if (existingSubscriptions > 0) {
-			throw new Meteor.Error('FEDERATION_Error_user_is_federated_on_rooms');
-		}
+	if (isUserFederated(user)) {
+		throw new Meteor.Error('error-user-is-federated', 'Cannot delete federated user');
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/lib/server/functions/deleteUser.ts:27]: I remember raising some concerns about this on the original PR but don't recall what came of it. Why is this even a requirement?

File: apps/meteor/app/mentions/server/server.js [hunk 2]
```diff
@@ -28,7 +28,20 @@ export class MentionQueries {
 	}
 
 	getChannels(channels) {
-		return Rooms.find({ name: { $in: [...new Set(channels)] }, t: { $in: ['c', 'p'] } }, { fields: { _id: 1, name: 1 } }).fetch();
+		return Rooms.find(
+			{
+				$and: [
+					{
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/mentions/server/server.js:34]: This first `$and` is not doing anything.

File: apps/meteor/app/models/server/models/Rooms.js [hunk 3]
```diff
@@ -344,13 +344,24 @@ export class Rooms extends Base {
...
+	findOneByNameAndType(name, type, options, includeFederatedRooms = false) {
 		const query = {
-			name,
 			t: type,
 			teamId: {
 				$exists: false,
 			},
+			...(includeFederatedRooms
+				? {
+						$and: [
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/models/server/models/Rooms.js:355]: This first `$and` is not doing anything

File: apps/meteor/app/models/server/models/Rooms.js [hunk 4]
```diff
@@ -344,13 +344,24 @@ export class Rooms extends Base {
...
+							{
+								$or: [
+									{ $and: [{ $or: [{ federated: { $exists: false } }, { federated: false }], name }] },
+									{ federated: true, fname: name },
+								],
+							},
+						],
+				  }
+				: { $and: [{ $or: [{ federated: { $exists: false } }, { federated: false }], name }] }),
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/models/server/models/Rooms.js:364]: This `$and` is not doing anything

File: apps/meteor/app/ui-message/client/popup/messagePopupConfig.js [hunk 5]
```diff
@@ -382,6 +383,7 @@ Template.messagePopupConfig.helpers({
 					.find(
 						{
 							name: exp,
+							federated: { $exists: false },
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/ui-message/client/popup/messagePopupConfig.js:386]: shouldn't this also check for `federation: false` ?

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]

File: apps/meteor/app/lib/server/functions/deleteUser.ts (hunk 1 above)
Comment (MarcosSpessatto) [reply apps/meteor/app/lib/server/functions/deleteUser.ts:27]: Yes, I remember, we've agreed we should discuss that internally and find a solution to this, having in mind GDPR rules, but it was not discussed yet, would you suggest removing this validation instead?

File: apps/meteor/app/mentions/server/server.js (hunk 2 above)
Comment (MarcosSpessatto) [reply apps/meteor/app/mentions/server/server.js:34]: Thanks, didn't notice that 😞

File: apps/meteor/app/lib/server/functions/deleteUser.ts (hunk 1 above)
Comment (MarcosSpessatto) [reply apps/meteor/app/lib/server/functions/deleteUser.ts]: @pierre-lehnen-rc I'm reverting this change for now.

**Review by pierre-lehnen-rc — APPROVED** [review]

**Review by hugocostadev — APPROVED** [review]

**Review by pierre-lehnen-rc — APPROVED** [review]

**Pull request merged**